components/brave_extension/extension/brave_extension/background/webDiscoveryProject.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { App } from '../../../../../vendor/web-discovery-project/build'
+
+declare var window: any
+
+if (App !== undefined) {
+  const APP = new App({
+    version: chrome.runtime.getManifest().version
+  })
+  window.WDP = APP
+
+  const WEB_DISCOVERY_PREF_KEY = 'brave.web_discovery_enabled'
+
+  // Dynamically inject WDP content script so that users with the pref disabled
+  // don't need to pay the loading cost on each page.
+  chrome.tabs.onUpdated.addListener((tabId: number, changeInfo: chrome.tabs.TabChangeInfo, tab: chrome.tabs.Tab) => {
+    if (tab.id) {
+      const isEnabled = APP.modules['web-discovery-project'].isEnabled
```

Comment:
Here we need to check if App has been initialized and is running instead. Since the `isEnabled` will just return true all the time (it only means that this module should be enabled by App).
```suggestion
      const isEnabled = APP.isRunning
```

Reply:
done in https://github.com/brave/brave-core/pull/9381/commits/a8a70bcb289baa97a23b906c5fe4d8c2b0c3b730